core/server/web/site/app.js

```diff
@@ -49,6 +49,41 @@ module.exports = function setupSiteApp(options = {}) {
     // /public/ghost-sdk.js
     siteApp.use(shared.middlewares.servePublicFile('public/ghost-sdk.js', 'application/javascript', constants.ONE_HOUR_S));
     siteApp.use(shared.middlewares.servePublicFile('public/ghost-sdk.min.js', 'application/javascript', constants.ONE_YEAR_S));
+
+    // /public/members.js
+    siteApp.get('/public/members-theme-bindings.js',
+        shared.middlewares.labs('members'),
+        shared.middlewares.servePublicFile.createPublicFileMiddleware(
+            'public/members-theme-bindings.js',
+            'application/javascript',
+            constants.ONE_HOUR_S
+        )
+    );
+    siteApp.get('/public/members-theme-bindings.min.js',
+        shared.middlewares.labs('members'),
+        shared.middlewares.servePublicFile.createPublicFileMiddleware(
+            'public/members-theme-bindings.min.js',
+            'application/javascript',
+            constants.ONE_YEAR_S
+        )
+    );
+    siteApp.get('/public/members.js',
+        shared.middlewares.labs('members'),
+        shared.middlewares.servePublicFile.createPublicFileMiddleware(
+            'public/members.js',
+            'application/javascript',
+            constants.ONE_HOUR_S
+        )
+    );
+    siteApp.get('/public/members.min.js',
+        shared.middlewares.labs('members'),
+        shared.middlewares.servePublicFile.createPublicFileMiddleware(
+            'public/members.min.js',
+            'application/javascript',
+            constants.ONE_YEAR_S
+        )
+    );
+
```

Comment:
Sorry I didn't get why the existing `servePublicFile` middleware couldn't have been used here similar to how we are currently serving `ghost-sdk.js` and why this refactor is needed in both files? 🤔 
```
siteApp.use(shared.middlewares.labs('members'), shared.middlewares.servePublicFile('public/members.js', 'application/javascript', constants.ONE_HOUR_S));
```

Reply:
It's cool - I tried to kinda explain in the commit a bit, but basically the existing stuff is `siteApp.use`d right? So if we add that members labs middleware, it will apply to the ENTIRE site app, rather than just one route. By using `get` it is fixed to a route and we can apply middleware as we see fit.

Reply:
I think its possible to restrict `app.use` to particular routes as well instead of being global, `path` is just the first parameter before middleware - https://expressjs.com/en/api.html#app.use

Even with `get` the middleware could be used as is though?

```
siteApp.use('/public/members.min.js', shared.middlewares.labs('members'), shared.middlewares.servePublicFile('public/members.js', 'application/javascript', constants.ONE_HOUR_S));

```
OR
```
    siteApp.get('/public/members.min.js',
        shared.middlewares.labs('members'),
        shared.middlewares.servePublicFile(
            'public/members.min.js',
            'application/javascript',
            constants.ONE_YEAR_S
        )
    );
```

Reply:
I didn't know that about `app.use`! Though i'm not sure why we would prefer to `use` rather than `get` - seems like a misuse of the express methods :thinking: 

And yes, while you can technically use the `publicServeFile` middleware with `get` it is making the path check twice, which is bad for 1. making seemingly redundant checks, and 2. we now have two pieces of code running the same (we hope) logic, which could change slightly, leading to inconsistencies.

By exporting a generic middleware which doesn't care about the `req.url` we encourage using it but leaving all the route matching stuff to be handled by express, which is built and battle tested for that, and the middleware only does the custom stuff.

Can revert if you think it's better to leave as is?

Reply:
> Though i'm not sure why we would prefer to use rather than get - seems like a misuse of the express methods 

Yeah its definitely more for the cases of applying certain middleware on all routes of a particular path, than a single path like here. So if both of members script are put into a `members` folder, this could be probably used with `/public/members/` as the path in `use` method to apply to both.

> By exporting a generic middleware which doesn't care about the req.url we encourage using it but leaving all the route matching stuff to be handled by express, which is built and battle tested for that, and the middleware only does the custom stuff.

Sounds sane to me, I think though then its true for all the other public routes being served as well via `use` currently, as they could move to `get` as well with the new logic. Lets merge it for now though then, and we can update it later, probably good idea to leave a TODO ?